epi/epi_functions.py

```diff
@@ -0,0 +1,265 @@
+    # 1: <xlsx> checks
+    #   - based on date provided, find corresponding <xlsx> file. If missing yield
+    #     informative error if file is not found (directions on how to structure the
+    #     filename)
+    #   - check that tb4 sheet exists with informative message on how to format the
+    #     sheet name
+    # 2: data integrity check
+    #   - expect first column to be date and monotinically increasing/decreasing
+    #   - expect columns 2:11 to be counts for age groups 0-85+ for men
+    #   - expect columns 13:22 to be counts for age groups 0-85+ for women
+    #   - all values for columns 2:22 should be >= 0 or NA (empty)
+    # 3: missing dates
+    #   - if any missing dates, ask user if assuming NA is ok
+    #     - if ok, add missing dates NA for all columns
+    #     - if not, abort
+    # 4: calculate sums
+    #   - calculate cumulative sums per age group per gender (20 new variables)
+    #   - calculate row-wise age group sums (irrespective of gender) of cumulative sums
+    #     (10 new variables)
+    #   - calculate row-wise sums for each gender of cumulative sums (2 new variables)
+    #   - calculate grand row-sums for genders combined of cumulative sums (1 new
+    #     variable)
+    #
+    # output of the script is these new 34 variables (<new_csv>) and should not
+    # overwrite <original_csv>
+    #
+    # 5: diff
+    #   - create a log of differences between <original_csv> and new table created in 4
+    #   - create a heatmap showing potential differences between <original_csv> and
+    #     <new_csv>
+
+    # original_csv = "original_tb4.csv"
+    # new_csv = "latest_dump_tb4.csv"
+    # datum = "2020-10-22"
+
+    if not datum:
+        raise ValueError("You did not enter a date.")
+
+    input_datum = datetime.datetime.strptime(datum, "%Y-%m-%d").strftime("%d%m%Y")
+
+    xlsx = glob.glob(pathname=f"*{input_datum}.xlsx")
+
+    # 1: <xlsx> checks
+    #   - based on date provided, find corresponding <xlsx> file. If missing yield
+    #     informative error if file is not found (directions on how to structure the
+    #     filename)
+    #   - check that tb4 sheet exists with informative message on how to format the
+    #     sheet name
+    if len(xlsx) > 1:
+        raise FileExistsError(
+            f"Found more than one file with datum {input_datum} in its file name."
+        )
+
+    xlsx = xlsx[0]
+
+    if not os.path.isfile(xlsx):
+        raise FileNotFoundError(
+            f"Unable to find file with datum {input_datum} in its " f"name."
+        )
+
+    age_groups_male = [
+        "0-4m",
+        "5-14m",
+        "15-24m",
+        "25-34m",
+        "35-44m",
+        "45-54m",
+        "55-64m",
+        "65-74m",
+        "75-84m",
+        "85+m",
+    ]
+    age_grups_female = [
+        "0-4f",
+        "5-14f",
+        "15-24f",
+        "25-34f",
+        "35-44f",
+        "45-54f",
+        "55-64f",
+        "65-74f",
+        "75-84f",
+        "85+f",
+    ]
```

Comment:
Could these two be generated from `age_groups` (defined below)?

Reply:
+ age groups could be defined as a constant at the top of the file